sequencer.c

```diff
@@ -2032,6 +2039,20 @@ static int pick_commits(struct todo_list *todo_list, struct replay_opts *opts)
 			*end_of_arg = '\0';
 			res = do_exec(item->arg);
 			*end_of_arg = saved;
+
+                        /* Reread the todo file if it has changed. */
+                        if (res == 0) {
+                                struct stat st;
+                                if ((res = stat(get_todo_path(opts), &st)))
+                                        return res;
```

Comment:
Oh, I see that there is one more thing the code does afterwards: it increments `todo_list->current`. And only then it checks whether `res` is non-zero. Maybe we should not return here but use

```c
if (stat(get_todo_path(opts), &st))
    res = error_errno(_("could not re-read '%s'"), get_todo_path(opts));
else if (match_stat_data(&todo_list->stat, &st)) {
    todo_list_release(todo_list);
    if (read_populate_todo(todo_list, opts))
        res = -1; /* error already reported */
    todo_list->current = -1; /* will be incremented below */
}
```